homeassistant/components/history/__init__.py

```diff
@@ -393,7 +393,7 @@ def entity_filter(self):
         if includes and not excludes:
             return or_(*includes)
 
-        if not excludes and includes:
+        if not includes and excludes:
```

Comment:
@bdraco you wrote this original code. It looks like this is a bug in the if-check.

The runtime effect is that previously `or_(*includes)` would translate to `or_(*[])` which is equivalent to `or_(False, *args)` according to the SQLALchemy error.

Reply:
Looks like a bug to me as well